src/swarm/util/log/ClientStats.d

```diff
@@ -146,6 +152,40 @@ public class ClientStats : StatsLog
     }
 
 
+    /***************************************************************************
+
+        Construct the class to use the default StatsLog appender, registers it
+        with the applications' reopenable files extension (if present), and
+        starts the update timer.
+
+        Params:
+            app = application class
+            epoll = epoll select dispatcher to register timer with
+            file_name = log file to write to
+            period = seconds delay between log writes
+
+    ***************************************************************************/
+
+    public this ( IApplication app, EpollSelectDispatcher epoll,
+        istring file_name, time_t period = default_period )
+    {
+        Appender newAppender ( istring file, Appender.Layout layout )
+        {
+            auto stream = new File(file, File.WriteAppending);
+
+            if ( auto reopenable_files_ext =
+                (cast(Application)app).getExtension!(ReopenableFilesExt) )
```

Comment:
This is a bit strange - why take `IApplication` only to immediately cast it to`Application`? Why not require `Application` argument at once?

Reply:
Yeah true. From the POV of the normal usage, either would be fine, but accepting `Application` directly simplifies this code. Will update.

Reply:
(I copied some of this code from another location.)